tests/HEDA.robot

```diff
@@ -0,0 +1,266 @@
+
+API Create Program
+    [Arguments]       ${record_type}=Academic_Program
+    ${prog_name} =    Generate Random String
+    ${rt_id} =        Get Record Type Id  Account  Academic_Program
+    ${prog_id} =      Salesforce Insert  Account
+    ...                   Name=${prog_name}
+    ...                   RecordTypeId=${rt_id}
+    &{program} =      Salesforce Get  Account  ${prog_id}
+    [return]          &{program}
+
+
+Create Course Enrollment
+    [Arguments]                ${contact_id}  ${offering_name}
+    Go To Record Home          ${contact_id}
+    Click Related List Button  Course Connections  New
+    Select Record Type         Student
+    Populate Lookup Field      Course Offering ID  ${offering_name}
+    Click Modal Button         Save
+    Wait Until Modal Is Closed
+    @{records} =               Salesforce Query  Course_Enrollment__c
+    ...                            Contact__c=${contact_id}
+    ...                            Course_Offering__r.Name=${offering_name}
+    &{enrollment} =            Get From List  ${records}  0
+    Store Session Record       Course_Enrollment__c  &{enrollment}[Id]
+
+Create Department
+    ${department_name} =  Generate Random String
+    Go To Object Home     Account
+    Click Object Button   New
+    Select Record Type    University Department
+    Populate Form         Account Name=${department_name}
+    Click Modal Button    Save
+    Wait Until Modal Is Closed
+    ${department_id} =    Get Current Record Id
+    Store Session Record  Account  ${department_id}
+    [return]              ${department_id}
+
+Create Program
+    ${program_name} =     Generate Random String
+    Go To Object Home     Account
+    Click Object Button   New
+    Select Record Type    Academic Program
+    Populate Form         Account Name=${program_name}
+    Click Modal Button    Save
+    Wait Until Modal is Closed
+    ${program_id} =       Get Current Record Id
+    Store Session Record  Account  ${program_id}
+    [return]              ${program_id}
+
+API Create Contact
+    ${first_name} =  Generate Random String
+    ${last_name} =   Generate Random String
+    ${contact_id} =  Salesforce Insert  Contact
+    ...                  FirstName=${first_name}
+    ...                  LastName=${last_name}
+    &{contact} =     Salesforce Get  Contact  ${contact_id}
+    [return]         &{contact}
+  
+Create Contact
+    ${first_name} =           Generate Random String
+    ${last_name} =            Generate Random String
+    Go To Object Home         Contact
+    Click Object Button       New
+    Populate Form
+    ...                       First Name=${first_name}
+    ...                       Last Name=${last_name}
+    Click Modal Button        Save    
+    Wait Until Modal Is Closed
+    ${contact_id} =           Get Current Record Id
+    Store Session Record      Contact  ${contact_id}
+    [return]                  ${contact_id}
+    
+Create Contact with Email
```

Comment:
This keyword could be removed if you accept &{fields} on Create Contact instead.  You'd call Create Contact like this:
```
Create Dictionary  Work Email=someemail@test.com
Create Contact  &{fields}
```

Also, I'd recommend not using your real email address in the tests :)